rustls/src/client/mod.rs

```diff
@@ -153,18 +153,25 @@ impl ClientConfig {
 
     /// Sets MTU to `mtu`.  If None, the default is used.
     /// If Some(x) then x must be greater than 5 bytes.
-    pub fn set_mtu(&mut self, mtu: &Option<usize>) {
+    pub fn set_mtu(&mut self, mtu: &Option<usize>) -> Result<(), Error> {
         // Internally our MTU relates to fragment size, and does
         // not include the TLS header overhead.
         //
         // Externally the MTU is the whole packet size.  The difference
         // is PACKET_OVERHEAD.
-        if let Some(x) = *mtu {
-            use crate::msgs::fragmenter;
-            debug_assert!(x > fragmenter::PACKET_OVERHEAD);
-            self.mtu = Some(x - fragmenter::PACKET_OVERHEAD);
-        } else {
-            self.mtu = None;
+        match *mtu {
+            Some(x) => {
+                use crate::msgs::fragmenter;
+                if let Some(m) = x.checked_sub(fragmenter::PACKET_OVERHEAD + 1) {
+                    self.mtu = Some(m + 1);
+                    return Ok(());
+                }
+                Err(Error::MtuError)
+            }
+            None => {
+                self.mtu = None;
+                Ok(())
+            }
         }
```

Comment:
I feel like this would be easier to follow, what do you think?

```rust
        self.mtu = (*mtu)
            .map(|mtu| {
                use crate::msgs::fragmenter;
                match mtu.checked_sub(fragmenter::PACKET_OVERHEAD + 1) {
                    Some(mtu) => Ok(mtu + 1),
                    None => Err(Error::MtuError),
                }
            })
            .transpose()?;
        Ok(())
```

Reply:
Or alternatively, something like this if the combinator usage is a bit much:

```rust
        let mtu = match mtu {
            Some(mtu) => *mtu,
            None => {
                self.mtu = None;
                return Ok(());
            }
        };

        use crate::msgs::fragmenter;
        let mtu = mtu
            .checked_sub(fragmenter::PACKET_OVERHEAD + 1)
            .ok_or(Error::MtuError)?;
        self.mtu = Some(mtu + 1);
        Ok(())
```